### PR TITLE
[fix](array/map) fix resize impl in array/map (#41595)

### PR DESCRIPTION
**Pull request by amorynan**
backport: https://github.com/apache/doris/pull/41595
this resize function called shrink function which defined in IColumn, and function set_num_rows in Block.cpp call shrink to make column rows changed, maybe less or more.
In this situation if we would make less rows which like cut the rows, so we should also make nested data in array/map resize to the less number of rows , otherwise we will meet exception

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: regression-test/suites/datatype_p0/nested_types/query/test_nested_type_with_resize.groovy [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
... """ /*set ShuffleSendBytes=0|ShuffleSendRows=0|FuzzyVariables=batch_size=4064,broker_load_batch_size=16352,disable_streaming_preaggregations=false,enable_distinct_streaming_aggregation=true,parallel_fragment_exec_instance_num=3,parallel_pipeline_task_num=5,profile_level=1,enable_pipeline_engine=true,enable_parallel_scan=true,parallel_scan_max_scanners_count=48,parallel_scan_min_rows_per_scanner=16384,parallel_prepare_threshold=13,enable_fold_constant_by_be=true,enable_rewrite_element_at_to_slot=true,runtime_filter_type=12,enable_parallel_result_sink=true,sort_phase_num=0,rewrite_or_to_in_predicate_threshold=100000,enable_function_pushdown=false,enable_common_expr_pushdown=true,enable_local_exchange=false,partitioned_hash_join_rows_threshold=1048576,partitioned_hash_agg_rows_threshold=1048576,partition_pruning_expand_threshold=10,enable_share_hash_table_for_broadcast_join=false,enable_two_phase_read_opt=true,enable_delete_sub_predicate_v2=false,enable_sort_spill=false,enable_agg_spill=false,enable_force_spill=false,data_queue_max_blocks=1,spill_streaming_agg_mem_limit=268435456,external_agg_partition_bits=5*/ select col22 from test_array_resize order by col1,col2,col3 limit 10; """
```
Comment (superdiaodiao) [inline regression-test/suites/datatype_p0/nested_types/query/test_nested_type_with_resize.groovy:69]: nit:
What about moving the common setting part outside the queries?

Comment (amorynan) [reply regression-test/suites/datatype_p0/nested_types/query/test_nested_type_with_resize.groovy:69]: I think It doesn't matter for this case

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**